commit 6d58497dd1d68231af201cf2a07bb9881c4efd2f
Author: Tanu Kaskinen <tanuk@iki.fi>
Date:   Thu Oct 6 23:28:37 2011 +0300

    daemon: Don't treat it as a fatal error if we can't connect to the session bus
    
    http://lists.freedesktop.org/archives/pulseaudio-discuss/2011-June/010276.html

diff --git a/src/daemon/main.c b/src/daemon/main.c
index e2e4a81..c32f7c7 100644
--- a/src/daemon/main.c
+++ b/src/daemon/main.c
@@ -1107,14 +1107,14 @@ int main(int argc, char *argv[]) {
 
 #ifdef HAVE_DBUS
     if (!conf->system_instance) {
-        if (!(server_lookup = pa_dbusobj_server_lookup_new(c)))
-            goto finish;
-        if (!(lookup_service_bus = register_dbus_name(c, DBUS_BUS_SESSION, "org.PulseAudio1")))
-            goto finish;
+        if ((server_lookup = pa_dbusobj_server_lookup_new(c))) {
+            if (!(lookup_service_bus = register_dbus_name(c, DBUS_BUS_SESSION, "org.PulseAudio1")))
+                goto finish;
+        }
     }
 
-    if (start_server && !(server_bus = register_dbus_name(c, conf->system_instance ? DBUS_BUS_SYSTEM : DBUS_BUS_SESSION, "org.pulseaudio.Server")))
-        goto finish;
+    if (start_server)
+        server_bus = register_dbus_name(c, conf->system_instance ? DBUS_BUS_SYSTEM : DBUS_BUS_SESSION, "org.pulseaudio.Server");
 #endif
 
 #ifdef HAVE_FORK
diff --git a/src/daemon/server-lookup.c b/src/daemon/server-lookup.c
index 45796e7..9bdc699 100644
--- a/src/daemon/server-lookup.c
+++ b/src/daemon/server-lookup.c
@@ -485,7 +485,7 @@ pa_dbusobj_server_lookup *pa_dbusobj_server_lookup_new(pa_core *c) {
     sl->path_registered = FALSE;
 
     if (!(sl->conn = pa_dbus_bus_get(c, DBUS_BUS_SESSION, &error)) || dbus_error_is_set(&error)) {
-        pa_log("Unable to contact D-Bus: %s: %s", error.name, error.message);
+        pa_log_warn("Unable to contact D-Bus: %s: %s", error.name, error.message);
         goto fail;
     }
 
